File: bifurcacion.cpp
#include<iostream>
#include<cmath>

using namespace std;

const double w0=1.0,var=0.2;
const double h=0.1,delta=1.0;
const double F=0.4,w=0.2;
const int N =1000;


double g( double theta0 , double omega0 , double t ) ;
double f( double theta0 , double omega0 , double t ) ;
int main( void ){

double omega=1;
double theta=0;
for (int i=0;i<N;i++) {
	cout<<i*h<<'\t'<<theta<<'\t'<<omega<<endl;
	omega = omega+h*f( theta , omega , i*h);
	theta = theta+h*g( theta , omega , i*h);
	}
	return 0;
}
double g ( double theta0 , double omega0 , double t ) {
	return omega0 ;}
double f ( double theta0 , double omega0 , double t ) {
	return -w0*w0*sin( theta0 ) - var*omega0 + F*sin( w*t+delta ) ;}

File: bifurcacion.py
import math as m

def g ( theta0 , omega0 ) :
	return omega0
def f ( theta0 , omega0 , t ) :
	return - w0 **2.* m . sin ( theta0 ) - gamma * omega0 + F * m . sin ( w * t + delta )
gamma =.2
w0 =1.
omega =1.
theta =0.
h =0.1
N =1000
delta =1.
w =.2
F =.4
for i in range ( N ) :
	print i *h , theta , omega
	omega = omega + h * f ( theta , omega , i * h )
	theta = theta + h * g ( theta , omega )
